matlab/CMakeLists.txt

```diff
@@ -32,8 +110,29 @@ endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake_modules)
 
-# Arrow is Required
-find_package(Arrow REQUIRED)
+# Only build the MATLAB interface C++ tests if MATLAB_BUILD_TESTS=ON.
+if(MATLAB_BUILD_TESTS)
+  # Should support custom GTEST_ROOT as well as package managers.
+  find_package(GTest)
+  if(NOT GTest_FOUND)
+    # Trigger an automatic build of the Arrow C++ libraries and bundled GoogleTest binaries.
+    build_arrow_and_gtest()
+  else()
+    # Should support custom ARROW_HOME as well as package managers.
+    find_package(Arrow)
+    if(NOT Arrow_FOUND)
+      # Trigger an automatic build of the Arrow C++ libraries.
+      build_arrow()
+    endif()
+  endif()
+else()
+  # Should support custom ARROW_HOME as well as package managers.
```

Comment:
`ARROW_HOME` is processed by `FindArrow.cmake`. So we don't need additional code here.

Reply:
See my comment above. The comment wording here is misleading. I'll reword this to be less ambigious.

Reply:
Fixed.